pkg/migration/controllers/resourcetransformation.go

```diff
@@ -0,0 +1,320 @@
+			log.TransformLog(transform).Errorf(message)
+			r.recorder.Event(transform,
+				v1.EventTypeWarning,
+				string(stork_api.ResourceTransformationStatusFailed),
+				message)
+			transform.Status.Status = stork_api.ResourceTransformationStatusFailed
+			err := r.client.Update(ctx, transform)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+	case stork_api.ResourceTransformationStatusInProgress:
+		err = r.validateTransformResource(ctx, transform)
+		if err != nil {
+			message := fmt.Sprintf("Error validating resource transformation specs: %v", err)
+			log.TransformLog(transform).Errorf(message)
+			r.recorder.Event(transform,
+				v1.EventTypeWarning,
+				string(stork_api.ResourceTransformationStatusFailed),
+				message)
+			transform.Status.Status = stork_api.ResourceTransformationStatusFailed
+			err := r.client.Update(ctx, transform)
+			if err != nil {
+				return err
+			}
+		}
+	case stork_api.ResourceTransformationStatusReady:
+	case stork_api.ResourceTransformationStatusFailed:
+		return nil
+	default:
+		log.TransformLog(transform).Errorf("Invalid status for ResourceTransformation: %v", transform.Status.Status)
+	}
+	return nil
+}
+
+func (r *ResourceTransformationController) validateSpecPath(transform *stork_api.ResourceTransformation) error {
+	for _, spec := range transform.Spec.Objects {
+		kind, err := getKind(spec.Resource)
+		if err != nil {
+			return err
+		}
+		if !resourcecollector.GetSupportedK8SResources(kind, []string{}) {
+			return fmt.Errorf("unsupported resource kind for transformation: %s", kind)
+		}
+		for _, path := range spec.Paths {
+			// TODO: this can be validated via CRDs as well, when we have defined schema
+			// for stork crds
+			if path.Operation == stork_api.JsonResourcePatch {
+				return fmt.Errorf("json patch for resources is not supported, operation: %s", path.Operation)
+			}
+			if !(path.Operation == stork_api.AddResourcePath || path.Operation == stork_api.DeleteResourcePath) {
+				return fmt.Errorf("unsupported resource patch operation given for kind :%s, operation: %s", kind, path.Operation)
+			}
+			if !(path.Type == stork_api.BoolResourceType || path.Type == stork_api.IntResourceType ||
+				path.Type == stork_api.StringResourceType || path.Type == stork_api.SliceResourceType ||
+				path.Type == stork_api.KeyPairResourceType) {
+				return fmt.Errorf("unsupported type for resource %s, path %s, type: %s", kind, path.Path, path.Type)
+			}
+		}
+	}
+	return nil
+}
+
+func (r *ResourceTransformationController) validateTransformResource(ctx context.Context, transform *stork_api.ResourceTransformation) error {
+	resourceCollectorOpts := resourcecollector.Options{}
+	for _, spec := range transform.Spec.Objects {
+		gvk := strings.Split(spec.Resource, "/")
+		if len(gvk) != 3 {
+			return fmt.Errorf("invalid resource type should be in format <group>/<version>/<kind>, actual: %s", gvk)
```

Comment:
nit: inconsistent Capitalization at beginning of errors, Above you use Invalid, unsupported, Unsupported. Make all Capital

Reply:
+1